Add note about collection element order (#42704)
quarkusio/quarkus: docs/src/main/asciidoc/config-mappings.adoc

configuration values in mapping groups. For collection with simple element types like `String`, their configuration
value is a comma separated string.

> NOTE: Note that only the `List` mapping is able to maintain element order. Hence, with `Set` mappings the element order is not maintained from the configuration files but is random.

Reviewer: ```suggestion
NOTE: Only the `List` mapping can maintain element order. Hence, with `Set` mappings the element order is not maintained from the configuration files but is random.
```
